### PR TITLE
Fix .travis.yml and composer.json for individual bundles and components

### PR DESCRIPTION
**Pull request by pjedrzejewski**

### CODE REVIEW

**Comment by bendavies** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
@Sylius/core-team Please review and merge if good to go, I'd like to run the subtree split manually and see how many builds are left to fix. :)

File: src/Sylius/Bundle/AddressingBundle/.travis.yml [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 
 php:
-  - 5.3
-  - 5.4
   - 5.5
+  - 5.6
+  - 7.0
 
-before_script: composer install --prefer-source --no-interaction
+before_script: composer install --dev --prefer-source
```
Comment (pamil) [inline src/Sylius/Bundle/AddressingBundle/.travis.yml]: We can remove `--dev` as it is the default behaviour.

File: src/Sylius/Bundle/ArchetypeBundle/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+php:
+  - 5.5
+  - 5.6
+  - 7.0
+
+before_script: composer install --dev --prefer-source
+
+script: bin/phpspec run -fpretty --verbose
+
```
Comment (pamil) [inline src/Sylius/Bundle/ArchetypeBundle/.travis.yml]: Extra blank line :)

File: src/Sylius/Bundle/ChannelBundle/.travis.yml [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-  - 5.3
-  - 5.4
   - 5.5
+  - 5.6
+  - 7.0
 
-before_script: composer install --prefer-source --no-interaction
+before_script: composer install --dev --prefer-source
 
 script: bin/phpspec run -fpretty --verbose
```
Comment (pamil) [inline src/Sylius/Bundle/ChannelBundle/.travis.yml]: At the most of `.travis.yml` there are doubled lines at the end

File: src/Sylius/Bundle/ProductBundle/Behat/ProductContext.php [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
 use Behat\Gherkin\Node\TableNode;
 use Doctrine\Common\Collections\ArrayCollection;
-use Sylius\Bundle\AttributeBundle\AttributeType\CheckboxAttributeType;
+use Sylius\Component\Attribute\AttributeType\CheckboxAttributeType;
```
Comment (pamil) [inline src/Sylius/Bundle/ProductBundle/Behat/ProductContext.php]: That's neither Travis nor Composer :D

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ProductBundle/Behat/ProductContext.php]: Yeah, I should mention that I also fix bad coupling between bundles/components where possible. :P In order to fix their builds.

File: src/Sylius/Bundle/AddressingBundle/.travis.yml (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/.travis.yml]: Why removing `--no-interaction`?

**Comment by michalmarcinkowski** [comment]
:+1:

File: src/Sylius/Component/Resource/composer.json [hunk 5]
```diff
@@ -26,10 +26,11 @@
         "doctrine/common": "~2.3",
         "symfony/event-dispatcher": "^2.7",
         "symfony/property-access": "^2.7",
-        "winzou/state-machine": "~0.1"
+        "winzou/state-machine": "~0.1",
+        "white-october/pagerfanta-bundle": "^1.0"
```
Comment (ogizanagi) [inline src/Sylius/Component/Resource/composer.json]: Why requiring the bundle instead of the component ?
Is it a hard dependency ?

Comment (pjedrzejewski) [inline src/Sylius/Component/Resource/composer.json]: Good catch, probably wrong copy paste, I will check but I am 99% sure we don't need a bundle here.

**Pull request merged**